### PR TITLE
multi: add ability to fund+use musig2 channels that commit to a tapscript root

### PR DESCRIPTION
**Pull request by Roasbeef**
In this PR, we add the initial plumbing that'll allow users to create and use musig2 channels that also commit to a tapscript root. Along the way, we start to store a new optional tapscript root in the `channeldb.OpenChannel` struct within the existing TLV stream appended to the older hard coded byte stream. The musig2 session itself will now conditionally pass in a tapscript root to the key aggregation and signing operations. We've also hooked up the lower half of the funding flow (lnwallet reservations) as well to ensure the new field gets propagated all the way down the relevant call stacks.

### CODE REVIEW

File: lnwallet/reservation.go [hunk 1]
```diff
@@ -212,6 +213,11 @@ type ChannelReservation struct {
 	// commitment state.
 	pushMSat lnwire.MilliSatoshi
 
+	// tapscriptRoot is the root of the tapscript tree that will be used to
+	// create the musig2 funding output. This is only used for taproot
+	// channels.
+	tapscriptRoot fn.Option[chainhash.Hash]
```
Comment (guggero) [inline lnwallet/reservation.go:219]: Unused since it's already in `partialState`?

**Review by guggero — COMMENTED** [review]
Great work, can really see the pieces coming together now :100:

File: lnwallet/channel_test.go [hunk 2]
```diff
@@ -827,6 +833,16 @@ func TestForceClose(t *testing.T) {
 			anchorAmt:            anchorSize * 2,
 		})
 	})
+	t.Run("taproot with tapscript root", func(t *testing.T) {
```
Comment (GeorgeTsagk) [inline lnwallet/channel_test.go:836]: IIUC these tests only check that the aux tapscript root can co-exist within the channel, but doesn't use the tapscript root in any way

correct?

**Review by GeorgeTsagk — COMMENTED** [review]
had a look, beautiful usage of `fn` package

**Comment by guggero** [comment]
Replacement version of this PR with all current review comments applied and rebased on the `0-19-staging` branch created here: https://github.com/lightningnetwork/lnd/pull/8683

**Pull request closed without merging**